elf_reader.go

```diff
@@ -325,6 +326,17 @@ func (ec *elfCode) loadProgramSections() (map[string]*ProgramSpec, error) {
 
 		progType, attachType, progFlags, attachTo := getProgType(sec.Name)
 
+		kernelModule := ""
+		if attachTo != "" {
+			if progType == Kprobe ||
```

Comment:
That makes sense. I did it here so the user would know the kernel module name if needed to load non-default BTF (like from BTFHub). It seems we may have to place that burden upon the user to correctly provide a kernel module name, if required. I think we can provide a helper function for determining if a function is within a kernel module, but I don't see a way to make it transparent AND also support non-default BTF since that is supplied during kernel load.